src/fiber/fiber.mli

```diff
@@ -228,20 +227,68 @@ end
 with type 'a fiber := 'a t
 
 module Mutex : sig
-  type 'a fiber = 'a t
+  type 'a fiber := 'a t
 
   type t
 
   val create : unit -> t
 
   val with_lock : t -> (unit -> 'a fiber) -> 'a fiber
 end
-with type 'a fiber := 'a t
+
+(** Dealing with cancellation *)
+module Cancel : sig
+  (** Fiber cancellation provides a way to cancel a running fiber. The
+      cancellation mechanism is not builtin the core of fiber and it is not
+      possible to cancel an arbitrary fiber.
+
+      Instead, cancellation is fully explicit. A careful long running fiber may
+      explicitely support cancellation by checking at strategic points if it has
+      been cancelled or not and reacting accordingly. *)
+
+  type 'a fiber := 'a t
+
+  (** A cancellation token. *)
+  type t
+
+  val create : unit -> t
+
+  (** Mark the given cancellation token as cancelled. *)
+  val cancel : t -> unit fiber
```

Comment:
We should clarify what happens on double cancellation? Is it an error? Is it idemptotent (my preference)?